db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/remove_from_provider.rb

```diff
@@ -9,22 +9,24 @@
 
 removal_type = $evm.inputs['removal_type'].downcase
 $evm.set_state_var('vm_removed_from_provider', false)
+ems = vm.ext_management_system
 
-if vm
-  ems = vm.ext_management_system
-  case removal_type
-  when "remove_from_disk"
-    if vm.miq_provision || vm.tagged_with?(category, tag)
-      $evm.log('info', "Removing VM:<#{vm.name}> from provider:<#{ems.try(:name)}>")
-      vm.remove_from_disk(false)
-      $evm.set_state_var('vm_removed_from_provider', true)
-    end
-  when "unregister"
-    $evm.log('info', "Unregistering VM:<#{vm.name}> from provider:<#{ems.try(:name)}")
-    vm.unregister
+if vm.nil? || ems.nil?
+  exit MIQ_OK
```

Comment:
We should definitely be logging something here before exiting so we know the condition that caused it.